packages/overlay/src/vaadin-overlay-position-mixin.js

```diff
@@ -233,8 +233,10 @@ export const PositionMixin = (superClass) =>
       const targetRect = this.positionTarget.getBoundingClientRect();
 
       if (targetRect.width === 0 && targetRect.height === 0 && this.opened) {
-        this.opened = false;
+        this.style.display = 'none';
         return;
+      } else if (this.style.display === 'none' && this.opened) {
```

Comment:
Note: there is no need to check for `this.opened` being true as there is a check above at the beginning of `_updatePosition()` with `return`.

Reply:
Good catch. Removed it also from the `if` clause.